samples/vision/face_samples.py

```diff
@@ -1,80 +1,694 @@
-import os.path
-
 from azure.cognitiveservices.vision.face import FaceClient
-from azure.cognitiveservices.vision.face.models import FaceAttributeType, Gender
 from msrest.authentication import CognitiveServicesCredentials
+from azure.cognitiveservices.vision.face.models import FaceAttributeType, HairColorType, TrainingStatusType, Person
+import os
+import uuid
+import time
 
 SUBSCRIPTION_KEY_ENV_NAME = "FACE_SUBSCRIPTION_KEY"
 FACE_LOCATION = os.environ.get("FACE_LOCATION", "westcentralus")
 
 IMAGES_FOLDER = os.path.join(os.path.dirname(os.path.realpath(__file__)), "images", "Face")
 
-def face_detect(subscription_key):
-    """ImageAnalysisInStream.
+def detect_faces(face_client, image_url):
+    """DetectFaces.
+
+    This will detect the faces found in the image with url image_url using the provided FaceClient instance and return a list of faces detected in the image.
+    """
+
+    detected_faces = face_client.face.detect_with_url(image_url, True, False, None, None, False)
+    if detected_faces is None or len(detected_faces) == 0:
+        raise Exception('No face detected from image {}'.format(image_url))
+    print("{} faces detected from image {}".format(len(detected_faces), image_url))
+    if detected_faces[0] is None:
+        raise Exception("Parameter return_face_id of detect_with_stream or detect_with_url must be set to true (by default) for recognition purpose.")
+    return list(detected_faces)
+
+
+
+def face_detection(subscription_key):
+    """FaceDetection.
+
+    This will print out all of the facial attributes for a list of images.
+    """
+
+    '''
+    Helper function for face_detection sample that returns a string representation of a person's accessories.
+
+    Parameter accessories: the accessories detected in an image.
+    Return: string representation of a person's accessories.
+    '''
+
+    def get_accessories(accessories):
+        if len(accessories) == 0:
+            return "No accessories"
+        accessory_array = []
+
+        for i in range(len(accessories)):
+            accessory_array.append(str(accessories[i]))
+        return ",".join(accessory_array)
+
+    '''
+    Helper function for face_detection sample that determines the emotion a person is showing.
+
+    Parameter emotion: the emotion object detected in an image.
+    Return: the emotion that the face is showing in the image.
+    '''
+
+    def get_emotion(emotion):
+        emotion_type = ""
+        emotion_value = 0.0
+        if emotion.anger > emotion_value:
```

Comment:
I would not do it that way. Loop on `emotion` attribute and keep the max
Peusdo code:
```python
max_emotion_value = 0.0
emotion_type = None
for emotion_attr in dir(emotion):
    emotion_value = getattr(emotion, emotion_attr)
    if emotion_value > max_emotion_value:
        max_emotion_value = emotion_value
        emotion_type = emotion_attr
return emotion_type
```
or
```python
emotions = [(emotion_attr, getattr(emotion, emotion_attr)) for emotion_attr in dir(emotion)]
emotions.sort()
return emotions[-1][0]  # The last one is the biggest, and the first element in the tuple is the string
```